components/sup/src/ctl_gateway/mod.rs

```diff
@@ -190,37 +188,54 @@ impl Write for CtlRequest {
         self.reply_partial(msg);
 
         // Down here, however, we're doing double-duty by *also*
-        // sending the output to the Supervisor's output stream. In
-        // _this_ case, we want to honor whatever global logging flags
-        // have been set; in particular, if we're emitting
-        // JSON-formatted logs or if we've been instructed to not
-        // output ANSI color codes, we need to strip those codes out.
+        // sending the output to the Supervisor's output stream.
         //
         // TODO (CM): Obviously, this feels hacky to the extreme. It'd
         // be nice to find a cleaner way to model this, but the fact
         // that CtlRequest is sending output to two destinations with
         // different formatting requirements complicates things a bit.
-        //
-        // TODO (MW): For `hab sup run` scenarios, it would be nice to
-        // support color on older versions of Windows. We could refactor
-        // outputln! to support a cross platform ColorWriter similar to
-        // what we use in common::UI.
-        let mut ansi_buffer = Buffer::ansi();
-        if let Some(spec) = &self.current_color_spec {
-            ansi_buffer.set_color(&spec)?;
-        }
-        ansi_buffer.write_all(buf)?;
-        ansi_buffer.flush()?;
-        let line = String::from_utf8_lossy(ansi_buffer.as_slice()).to_string();
 
-        let maybe_stripped = if output::is_json() || !output::is_color() {
-            STRIP_ANSI_CODES.replace_all(&line, "")
+        let is_line_ending = line.ends_with('\n');
+        if self.is_new_line || output::is_json() {
+            let so = if let Some(spec) = &self.current_color_spec {
+                StructuredOutput::colored(
+                    PROGRAM_NAME.as_str(),
+                    LOGKEY,
+                    line!(),
+                    file!(),
+                    column!(),
+                    &line.trim_right_matches('\n'),
+                    spec.clone(),
+                )
+            } else {
+                StructuredOutput::new(
+                    PROGRAM_NAME.as_str(),
+                    LOGKEY,
+                    line!(),
+                    file!(),
+                    column!(),
+                    &line.trim_right_matches('\n'),
+                )
+            };
```

Comment:
We can eliminate the duplication here (also `.as_str()` can be replaced with just a borrow):
```rust
            let so = StructuredOutput::colored(
                &PROGRAM_NAME,
                LOGKEY,
                line!(),
                file!(),
                column!(),
                line.trim_right_matches('\n'),
                self.current_color_spec.clone().unwrap_or_default(),
            );
```